diesel/src/pg/expression/functions.rs

```diff
@@ -1423,3 +1423,56 @@ define_sql_function! {
     /// ```
     fn array_upper<Arr: ArrayOrNullableArray + SingleValue>(array: Arr, dimension: Integer) -> Nullable<Integer>;
 }
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// Converts any SQL value to json
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::to_json;
+    /// #     use serde_json::Value;
+    /// #     use diesel::sql_types::{Integer, Array, Json, Text, Nullable};
+    /// #     let connection = &mut establish_connection();
+    /// let result = diesel::select(to_json::<Integer, _>(1))
+    ///     .get_result::<Value>(connection)?;
+    ///
+    /// let expected: Value = serde_json::from_str("1").unwrap();
+    /// assert_eq!(expected, result);
+    ///
+    /// let result = diesel::select(to_json::<Array<Integer>, _>(vec![1, 2, 3]))
+    ///     .get_result::<Value>(connection)?;
+    ///
+    /// let expected: Value = serde_json::from_str("[1, 2, 3]").unwrap();
+    /// assert_eq!(expected, result);
+    ///
+    /// let result = diesel::select(to_json::<Array<Text>, _>(vec!["abcd", "xyz"]))
+    ///     .get_result::<Value>(connection)?;
+    ///
+    /// let expected: Value = serde_json::from_str(r#"["abcd", "xyz"]"#).unwrap();
+    /// assert_eq!(expected, result);
+    ///
+    /// let result = diesel::select(to_json::<Array<Nullable<Text>>, _>(Vec::<String>::new()))
+    ///     .get_result::<Value>(connection)?;
+    ///
+    /// let expected: Value = serde_json::from_str("[]").unwrap();
+    /// assert_eq!(expected, result);
+    ///
+    /// let result = diesel::select(to_json::<Nullable<Array<Text>>, _>(None::<Vec<String>>))
+    ///     .get_result::<Value>(connection);
+    ///
+    /// assert!(result.is_err());
+    ///
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn to_json<E: SingleValue>(e: E) -> Json;
```

Comment:
This function seems to return `null` if you pass in a null values. We want to have a doc test demonstrating that this is correctly propagated.

Reply:
ive changed the last test a bit, let me know if its okay